numpy/array_api/fft.py

```diff
@@ -0,0 +1,296 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Union, Optional, Literal
+
+if TYPE_CHECKING:
+    from ._typing import Device
+    from collections.abc import Sequence
+
+from ._dtypes import (
+    _floating_dtypes,
+    _real_floating_dtypes,
+    _complex_floating_dtypes,
+    float32,
+    complex64,
+)
+from ._array_object import Array
+from ._data_type_functions import astype
+
+import numpy as np
+
+def fft(
+    x: Array,
+    /,
+    *,
+    n: Optional[int] = None,
+    axis: int = -1,
+    norm: Literal["backward", "ortho", "forward"] = "backward",
+) -> Array:
+    """
+    Array API compatible wrapper for :py:func:`np.fft.fft <numpy.fft.fft>`.
+
+    See its docstring for more information.
+    """
+    if x.dtype not in _complex_floating_dtypes:
+        raise TypeError("Only complex floating-point dtypes are allowed in fft")
```

Comment:
In the 2022.12 version of the standard, it says the inputs should be floating point and outputs should be complex: https://data-apis.org/array-api/2022.12/extensions/generated/array_api.fft.fft.html

Reply:
(but I found the dtypes used in the test suite disagree with the dtypes in the documentation in many places)

Reply:
https://github.com/data-apis/array-api/pull/720

Reply:
And yes, the test suite for fft needs to be updated in many places. I'm not aware of other places outside of fft that are wrong, so if you find any please open issues for those.